lib/_http_common.js

```diff
@@ -214,7 +238,19 @@ const tokenRegExp = /^[\^_`a-zA-Z\-0-9!#$%&'*+.|~]+$/;
  * @returns {boolean}
  */
 function checkIsHttpToken(val) {
-  return tokenRegExp.test(val);
+  if (val.length >= 10) {
```

Comment:
We can probably safely increase this to 30. There are a fair number of common header names that are over 10 characters (e.g. `content-length`, `authorization`, `content-type`, `content-security-policy`, etc)

Reply:
When looking at the benchmark results, it seems that the benefit over a regex is likely zero or negative for more than ten characters.

Reply:
@BridgeAR I think that's because the benchmarked code includes this check, so for 10+ chars it's not testing the new approach at all.